File: app-kotlin/src/main/java/home/self/beerviewer_mvvm/app_kotlin/data/source/remote/BeerRemoteDataSource.java
package home.self.beerviewer_mvvm.app_kotlin.data.source.remote;


import android.util.Log;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import home.self.beerviewer_mvvm.app_kotlin.data.model.BeerModel;
import home.self.beerviewer_mvvm.app_kotlin.data.source.BeerDataSource;
import home.self.beerviewer_mvvm.app_kotlin.network.BeerApiService;
import io.reactivex.Flowable;

@Singleton
public class BeerRemoteDataSource implements BeerDataSource {

    @Inject
    public BeerRemoteDataSource(BeerApiService apiService) {
        this.apiService = apiService;
    }

    private final BeerApiService apiService;

    @Override
    public void saveBeers(List<BeerModel> beers) { }

    @Override
    public Flowable<List<BeerModel>> getBeers() {
        return apiService.getBeers();
    }

    @Override
    public Flowable<List<BeerModel>> getBeers(int pageStart, int perPage) {
        Log.d("zxcv111", "a");
        return apiService.getBeers(pageStart, perPage);
    }

    @Override
    public Flowable<BeerModel> getBeer(int beerId) {
        return apiService.getBeer(beerId).toFlowable();
    }

    @Override
    public void saveBeer(BeerModel beer) { }
}

File: app-kotlin/src/main/java/home/self/beerviewer_mvvm/app_kotlin/data/source/remote/BeerRemoteDataSource.kt
package home.self.beerviewer_mvvm.app_kotlin.data.source.remote


import home.self.beerviewer_mvvm.app_kotlin.data.model.BeerModel
import home.self.beerviewer_mvvm.app_kotlin.data.source.BeerRepositoryApi
import home.self.beerviewer_mvvm.app_kotlin.network.BeerApiService
import io.reactivex.Flowable
import io.reactivex.Observable
import io.reactivex.Single
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
internal class BeerRemoteDataSource @Inject constructor(
        private val apiService: BeerApiService

) : BeerRepositoryApi {

    override fun fetchBeers(pageStart: Int, perPage: Int): Flowable<List<BeerModel>> = Flowable.just(mutableListOf())

    override fun fetchBeersFromRemote(pageStart: Int, perPage: Int): Single<List<BeerModel>> = apiService.fetchBeers(pageStart, perPage)

    override fun fetchBeer(beerId: Int): Flowable<BeerModel> = Flowable.just(BeerModel())

    override fun fetchBeerFromRemote(beerId: Int): Single<BeerModel> = apiService.fetchBeer(beerId)

    override fun saveBeer(beer: BeerModel) {}

    override fun getIndex(): Observable<Int> = Observable.just(1)

    override fun saveBeers(beers: List<BeerModel>) {}

}
